### PR TITLE
[DROOLS-4573] Cell selection doesn't span the entire cell

### PR DESCRIPTION
**Pull request by yesamer**
@dupliaka @gitgabrio @danielezonca can you please test and review it?

The original GridPanel fully supports cells selection span. I simply restored the native handling removing the wrong overridden logic in ScenarioSimulation code.

Screenshot from 2019-09-30 10-26-36

Bow works for:
- Left click,
- Right click
- Keyboard navigation
- Double click + editing mode

https://issues.jboss.org/browse/DROOLS-4573

### CODE REVIEW

**Comment by yesamer** [comment]
@dupliaka @gitgabrio @danielezonca done :)

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by dupliaka** [comment]
Still reproduces if i create a new column for Header

Steps to reproduce on **Test Scenario** Screen:
Given header >Right click> Insert RightMost Column

image

**Comment by yesamer** [comment]
@dupliaka Hi Anna! Thank you for reporting that case. I think this can be managed in a different ticket, then I kindly ask you to open a new ticket reporting this case and to continue to test this PR. Thank you!
@jomarko Can you please test the keyboard navigationas well?

**Comment by dupliaka** [comment]
> @dupliaka Hi Anna! Thank you for reporting that case. I think this can be managed in a different ticket, then I kindly ask you to open a new ticket reporting this case and to continue to test this PR. Thank you!
> @jomarko Can you please test the keyboard navigationas well?

https://issues.jboss.org/browse/DROOLS-4582

**Comment by dupliaka** [comment]
An Improvement not related to this feature [DROOLS-4586](https://issues.jboss.org/browse/DROOLS-4586)

**Comment by jomarko** [comment]
@yesamer we need to update [class](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandler.java), see the video [1]. Now user is not able to change instance alias by keyboard.

I am the author of the [class](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandler.java) so please tell me if I should create ticket for my sleft, or you will update as part of this PR, or what is your preference, thank you.

Otherwise the keyboard control seems fine.

[1]
https://drive.google.com/open?id=1bc6MnYI4cJr00j5R38yNFTXHTXQpOzw-

**Comment by yesamer** [comment]
@jomarko Changes added on my last commit, thank you for your suggestion!

**Comment by yesamer** [comment]
> @yesamer I am sorry, we need to update also [ScenarioInvokeContextMenuForSelectedCell](https://github.com/kiegroup/drools-wb/blob/master/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCell.java)
> 
> In the attached picture, user is not able to add that INSTANCE 7 column with keyboard.
> Screenshot from 2019-10-03 13-57-41

Thank you for reporting @jomarko, can you please tell me which are steps to add a column with keyboard navigation, in order to test it?

**Comment by jomarko** [comment]
@yesamer sure, it is again combination of keys **Ctrl+Enter**. Just prepare an instance header spanning multiple property headers, like the selected one in the attached picture and then try to press the given combination of keys.

**Comment by yesamer** [comment]
@jomarko Thank you Jozef, changes applied in my last commit.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.ext.wires.core.grids.client.model.GridColumn;
+import org.uberfire.ext.wires.core.grids.client.model.GridData;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.GridWidget;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+import static junit.framework.TestCase.assertTrue;
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java]: Please replace with import static org.junit.Assert.assertTrue;

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    public void isExecutable_CellsSelected() {
+        when(scenarioGridModelMock.getSelectedCells()).thenReturn(Lists.create(selectedCell, selectedCell2));
+        when(scenarioGridModelMock.getSelectedHeaderCells()).thenReturn(Collections.emptyList());
+        assertFalse(handler.isExecutable(scenarioGridMock));
+    }
+
+    @Test
+    public void isExecutable_HeaderCellSelected() {
+        when(scenarioGridModelMock.getSelectedCells()).thenReturn(Collections.emptyList());
+        when(scenarioGridModelMock.getSelectedHeaderCells()).thenReturn(Lists.create(selectedCell, selectedCell2));
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java]: just one cell should be here

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    public void isExecutable_CellsSelected() {
+        when(scenarioGridModelMock.getSelectedCells()).thenReturn(Lists.create(selectedCell, selectedCell2));
+        when(scenarioGridModelMock.getSelectedHeaderCells()).thenReturn(Collections.emptyList());
+        assertFalse(handler.isExecutable(scenarioGridMock));
+    }
+
+    @Test
+    public void isExecutable_HeaderCellSelected() {
+        when(scenarioGridModelMock.getSelectedCells()).thenReturn(Collections.emptyList());
+        when(scenarioGridModelMock.getSelectedHeaderCells()).thenReturn(Lists.create(selectedCell, selectedCell2));
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java]: just one cell please

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGrid;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridColumn;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.ext.wires.core.grids.client.model.GridData;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+import static junit.framework.TestCase.assertTrue;
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java]: import static org.junit.Assert.assertTrue;

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java (hunk 2 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java]: Done!

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java (hunk 1 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioInvokeContextMenuForSelectedCellTest.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java (hunk 3 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java (hunk 4 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/ScenarioSimulationKeyboardEditHandlerTest.java]: Done

**Pull request merged**